sale_loyalty_exclude/models/sale_order.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _get_no_effect_on_threshold_lines(self):
+        """
+        Returns the lines that have no effect on the minimum amount to reach
+        """
+        self.ensure_one()
+        lines = self.order_line.filtered(lambda line: line.product_id.loyalty_exclude)
+        return lines + super()._get_no_effect_on_threshold_lines()
```

Comment:
Don't you want an `union` here instead of `+`?

Reply:
I think union and + are equivalent in the case because this is a recordset

Reply:
![image](https://github.com/OCA/sale-workflow/assets/112623095/2e5d76e4-5ed3-40d7-b97b-11ac3312d873)
Apparently not

Reply:
![image](https://github.com/OCA/sale-workflow/assets/112623095/97f644e3-e5d9-40f3-b357-5742bf4111a8)
![image](https://github.com/OCA/sale-workflow/assets/112623095/7c5632f9-7251-402e-b25d-383c5bc44610)

Reply:
Thank you @Ricardoalso in this case is better to use `|`

Reply:
In standard 17.0 they are using `+` https://github.com/odoo/odoo/blob/11bd9e01b79ec94c6e8e637a30cd412f01f2d791/addons/sale_loyalty_delivery/models/sale_order.py#L25

Reply:
Changed